kin/blockchain/channel_manager.py

```diff
@@ -2,95 +2,84 @@
+    async def get_channel(self) -> str:
         """
         Get an available channel
-        :param float timeout: (Optional) How long to wait before raising an exception
+
         :return a free channel seed
-        :rtype str
 
-        :raises KinErrors.ChannelBusyError
         """
-        try:
-            channel = self.channel_pool.get(timeout=timeout)
-        except queue.Empty:
-            raise ChannelsBusyError()
+        channel = await self.channel_pool.get()
 
         try:
             yield channel
         finally:
-            if self.channel_pool.queue[channel] != ChannelStatuses.UNDERFUNDED:
-                self.put_channel(channel)
+            if self.channel_pool._queue[channel] != ChannelStatuses.UNDERFUNDED:
+                await self.put_channel(channel)
 
-    def put_channel(self, channel, timeout=CHANNEL_PUT_TIMEOUT):
+    async def put_channel(self, channel) -> None:
         """
         Set a channel status back to FREE
-        :param str channel: the channel to set back to FREE
-        :param float timeout: (Optional) How long to wait before raising an exception
 
-        :raises KinErrors.ChannelsFullError
+        :param str channel: the channel to set back to FREE
         """
-        try:
-            self.channel_pool.put(channel, timeout=timeout)
-        except queue.Full:
-            raise ChannelsFullError()
+        await self.channel_pool.put(channel)
 
-    def get_status(self, verbose=False):
+    def get_status(self, verbose: Optional[bool] = False) -> dict:
         """
         Return the current status of the channel manager
-        :param bool verbose: Include all channel seeds and their statuses in the response
+
+        :param verbose: Include all channel seeds and their statuses in the response
         :return: The status of the channel manager
-        :rtype dict
         """
         free_channels = len(self.channel_pool.get_free_channels())
         status = {
-            'total_channels': len(self.channel_pool.queue),
+            'total_channels': len(self.channel_pool._queue),
             'free_channels': free_channels,
-            'non_free_channels': len(self.channel_pool.queue) - free_channels
+            'non_free_channels': len(self.channel_pool._queue) - free_channels
         }
         if verbose:
-            status['channels'] = self.channel_pool.queue
+            status['channels'] = self.channel_pool._queue
 
         return status
 
 
 class ChannelStatuses(str, Enum):
     """Contains possible statuses for channels"""
+
     # subclass str to be able to serialize to json
     FREE = 'free'
     TAKEN = 'taken'
     UNDERFUNDED = 'underfunded'
 
 
-# TODO: remove object when we kill python2
-class ChannelPool(queue.Queue, object):
+class ChannelPool(queue):
     """
-    A thread-safe queue that sets a member's status instead of pulling it in/out of the queue.
+    An async queue that sets a member's status instead of pulling it in/out of the queue.
```

Comment:
out of curiosity, why queue is needed? seems simple list/set will provide the same functionality? I mean there's no real use of push/pop here.
(mainly trying to understand that I'm not missing something with my node impl for that)

Reply:
There is really no use of any push/pop here. I used the queue class before since it helped me with thread-safety. 

Now I'm inheriting it since it has asnyc methods to get/push back to the "queue". Otherwise, I would need to write my own implementation to select an item from the set asyncrounsly

Reply:
ok, so I understood it correctly, I would suggest to call it `pool` instead of `queue` (just in case it is a public api)